src/Chapter2.hs

```diff
@@ -728,7 +741,10 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate [] = []
+smartReplicate l = concat( map (\ (x) -> (replicate x x) ) l )
```

Comment:
Great! 👍🏼 
If you want to write this function even more elegantly, you can use the standard function `concatMap` which is a combination of `concat` and `map` as the name suggests 🙂
After that, you could notice, that you would be able to eta-reduce on the last argument :)

Reply:
Nice one, I hope I get to know soon the standard functions, I just have one doubt, what about the checks? I had to remove the "smartReplicate [] = []" because when eta reduce the arguments did not match (I see in this case was not necessary, but I guess maybe will be required in some other case)

Reply:
Yeah, for eta-reducing you indeed need to remove this extra check. Fortunately, it's not a problem, since both `map` and `concatMap` work perfectly on empty lists 👍🏻